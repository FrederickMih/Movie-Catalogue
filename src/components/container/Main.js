import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import MovieContainer from './MovieContainer';

import { fetchMovies } from '../../actions';

const Main = (props) => {
  useEffect(() => {
    props.fetchMovies(props.keyword);
  }, []);

  return (
    <div className="container">
      <h1>Helllloooooooooooo...............</h1>
      <MovieContainer />
    </div>
  );
};

const mapStateToProps = (state) => ({
  movie: state.movies,
});

Main.propTypes = {
  keyword: PropTypes.string,
  fetchMovies: PropTypes.func.isRequired,
};

Main.defaultProps = {
  keyword: '',
};

export default connect(mapStateToProps, { fetchMovies })(Main);
