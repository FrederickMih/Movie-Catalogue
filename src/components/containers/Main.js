import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import SearchForm from './SearchForm';
import MovieContainer from './MovieContainer';
import PageLoader from '../presentation/PageLoader';
import Pagination from '../presentation/Pagination';
import { fetchMovies } from '../../actions/index';
import '../../styles/main.css';

const Main = (props) => {
  const [page, setPage] = useState(1);
  const { loading } = props;

  useEffect(() => {
    props.fetchMovies(props.keyword, page);
  }, [page]);

  const handleNextClick = () => {
    setPage((page) => page + 1);
  };

  const handlePreviousClick = () => {
    setPage((page) => (page === 1 ? page : page - 1));
  };

  return (
    <div className="container movie-container">
      <SearchForm />
      {loading ? <PageLoader /> : <MovieContainer />}
      <Pagination handleNextClick={handleNextClick} handlePreviousClick={handlePreviousClick} />
    </div>
  );
};

const mapStateToProps = (state) => ({
  loading: state.movies.loading,
});

Main.propTypes = {
  loading: PropTypes.bool,
  keyword: PropTypes.string,
  fetchMovies: PropTypes.func.isRequired,
};

Main.defaultProps = {
  loading: false,
  keyword: false,
};

export default connect(mapStateToProps, { fetchMovies })(Main);
