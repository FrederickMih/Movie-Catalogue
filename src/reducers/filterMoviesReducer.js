import { FILTER_MOVIE } from '../actions/constants';

export const initialState = {
  filter: 'All',
};

const filterMovieReducer = (state = initialState, action) => {
  switch (action.type) {
    case FILTER_MOVIE: {
      const filter = action.payload;
      return {
        ...state, filter,
      };
    }
    default:
      return state;
  }
};

export default filterMovieReducer;
