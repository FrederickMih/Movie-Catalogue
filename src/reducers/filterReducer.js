import { FILTER_MOVIE } from '../actions';

export const initialState = {
  filter: 'Films',
};

const filterMovieReducer = (state = initialState, action) => {
  switch (action.type) {
    case FILTER_MOVIE: {
      const filter = action.payload;
      return {
        ...state, filter,
      };
    }
    default:
      return { ...state };
  }
};

export default filterMovieReducer;
