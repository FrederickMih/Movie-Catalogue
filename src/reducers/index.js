import { applyMiddleware, combineReducers, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import moviesReducer from './moviesReducers';
import searchLoadingReducer from './searchLoadingReducers';
import filterMovieReducer from './filterMoviesReducer';


const initialState = {};

// eslint-disable-next-line max-len
export const rootReducer = combineReducers({ movies: moviesReducer, searchLoadingReducer, filter: filterMovieReducer });

const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware()),
);

export default store;
