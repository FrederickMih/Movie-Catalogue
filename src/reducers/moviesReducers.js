import {
  FETCH_MOVIES,
  FETCH_MORE_MOVIES,
} from '../actions';

export const initialState = {
  keyword: '',
  movies: [],
  movie: [],
  loading: false,
  initialScreen: true,
  page: 1,
};

const moviesReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_MOVIES:
      return {
        ...state,
        movies: action.payload,
        loading: false,
      };
    case FETCH_MORE_MOVIES:
      return {
        ...state,
        movie: action.payload,
        loading: false,
      };
    default:
      return state;
  }
};

export default moviesReducer;
