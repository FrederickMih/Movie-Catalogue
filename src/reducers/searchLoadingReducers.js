import {
  SEARCH_MOVIE,
  LOADING,
} from '../actions';

export const initialState = {
  keyword: '',
  movies: [],
  movie: [],
  loading: false,
  page: 1,
};

const searchLoadingReducer = (state = initialState, action) => {
  switch (action.type) {
    case SEARCH_MOVIE:
      return {
        ...state,
        keyword: action.payload,
        loading: false,
      };
    case LOADING:
      return {
        ...state,
        loading: true,
      };
    default:
      return state;
  }
};

export default searchLoadingReducer;
