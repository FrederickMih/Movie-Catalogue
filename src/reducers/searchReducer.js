import {
  SEARCH_MOVIE, FETCH_MOVIES, FETCH_MOVIE_DETAILS, LOADING,
} from '../actions/constants';

export const initialState = {
  keyword: '',
  movies: [],
  loading: false,
  initialScreen: true,
  movie: [],
  page: 1,
};

const searchReducer = (state = initialState, action) => {
  switch (action.type) {
    case SEARCH_MOVIE:
      return {
        ...state,
        keyword: action.payload,
        loading: false,
      };
    case FETCH_MOVIES:
      return {
        ...state,
        movies: action.payload,
        loading: false,
      };
    case FETCH_MOVIE_DETAILS:
      return {
        ...state,
        movie: action.payload,
        loading: false,
      };
    case LOADING:
      return {
        ...state,
        loading: true,
      };
    default:
      return state;
  }
};

export default searchReducer;
