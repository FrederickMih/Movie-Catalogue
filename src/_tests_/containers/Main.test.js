import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import { composeWithDevTools } from 'redux-devtools-extension';
import Main from '../../components/containers/Main';
import { rootReducer } from '../../reducers/index';

const state = {};
const middleware = [thunk];
const store = createStore(rootReducer, state, composeWithDevTools(applyMiddleware(...middleware)));
const Wrapper = ({ children }) => (
  <Provider store={store}>{children}</Provider>
);

Wrapper.propTypes = {
  children: PropTypes.shape({}).isRequired,
};

describe('Main', () => {
  it('should renders correctly', () => {
    const tree = render(<Main />, { wrapper: Wrapper });
    expect(tree).toMatchSnapshot();
  });
});
