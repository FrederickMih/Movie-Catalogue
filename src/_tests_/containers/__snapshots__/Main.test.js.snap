// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container movie-container"
      >
        <div
          class="jumbotron jumbotron-fluid mt-5 text-center search-form-div"
        >
          <div
            class="container"
          >
            <form
              id="searchForm"
            >
              <input
                class="form-control me-2"
                name="searchText"
                placeholder=" Search for..."
                type="keyword"
              />
              <div>
                <select
                  aria-label="Default select example"
                  class="form-select my-5"
                >
                  <option
                    value="All"
                  >
                    All
                  </option>
                  <option
                    value="movie"
                  >
                    movie
                  </option>
                  <option
                    value="series"
                  >
                    series
                  </option>
                </select>
              </div>
              <button
                class="btn btn-dark w-25 btn-bg mb-3"
                type="submit"
              >
                Search
              </button>
            </form>
          </div>
        </div>
        <div
          class="row"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container movie-container"
    >
      <div
        class="jumbotron jumbotron-fluid mt-5 text-center search-form-div"
      >
        <div
          class="container"
        >
          <form
            id="searchForm"
          >
            <input
              class="form-control me-2"
              name="searchText"
              placeholder=" Search for..."
              type="keyword"
            />
            <div>
              <select
                aria-label="Default select example"
                class="form-select my-5"
              >
                <option
                  value="All"
                >
                  All
                </option>
                <option
                  value="movie"
                >
                  movie
                </option>
                <option
                  value="series"
                >
                  series
                </option>
              </select>
            </div>
            <button
              class="btn btn-dark w-25 btn-bg mb-3"
              type="submit"
            >
              Search
            </button>
          </form>
        </div>
      </div>
      <div
        class="row"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
